engine/src/main/java/org/terasology/rendering/dag/RenderGraph.java

```diff
@@ -62,11 +69,11 @@ public Node removeNode(SimpleUri nodeUri) {
 
         Node node = findNode(nodeUri);
```

Comment:
If findNode doesn't find the nodeUri in the map, the method returns null. In that context, there is no need to check if the graph contains the node: if findNode can't find it, neither will the graph. 

So, I'd add a null check on the result of findNode and I'd remove the check on the graph. The following check, with adjacentNodes, is fine though.